From dea0de8d414eac1e42a56ae44a4b666d58774dfd Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 22 Sep 2017 22:10:33 +0800
Subject: [PATCH 2582/5242] MLK-16536-8 video: fbdev: dcss: move 'ctxld_list'
 field to 'struct ctxld_fifo'

commit  90abbb4bf5161fbc86fc2f71ceb5701060685e37 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'ctxld_list' is more closely related to 'struct ctxld_info'
structure. So moving its definition to this structure is more
reasonable.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 30ad004..a616c2d 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -264,6 +264,7 @@ struct ctxld_fifo {
 	uint32_t size;
 	void *vaddr;
 	dma_addr_t dma_handle;
+	struct list_head ctxld_list;	/* manage context loader */
 	DECLARE_KFIFO_PTR(fifo, struct ctxld_unit);
 	struct scatterlist sgl[1];
 	uint32_t sgl_num;
@@ -316,7 +317,6 @@ struct dcss_info {
 	struct platform_device *pdev;
 	void __iomem *base;
 	void __iomem *blkctl_base;
-	struct list_head ctxld_list;	/* manage context loader */
 	spinlock_t llock;		/* list lock: for ctxld_list */
 	int irqs[DCSS_IRQS_NUM];
 	uint32_t irqs_num;
@@ -723,6 +723,7 @@ static int ctxld_fifo_alloc(struct device *dev,
 	/* TODO: sgl num can be changed if required */
 	cfifo->sgl_num = 1;
 
+	INIT_LIST_HEAD(&cfifo->ctxld_list);
 	init_waitqueue_head(&cfifo->cqueue);
 	init_completion(&cfifo->complete);
 
@@ -3131,7 +3132,6 @@ static int dcss_info_init(struct dcss_info *info)
 	int ret = 0;
 	struct platform_device *pdev = info->pdev;
 
-	INIT_LIST_HEAD(&info->ctxld_list);
 	spin_lock_init(&info->llock);
 
 	info->dcss_state = DCSS_STATE_RESET;
-- 
1.7.9.5

